### ISSUE TITLE
LockServiceImpl can allow locks to be held past expiration

### ISSUE THREAD

**Issue by nziebart** [palantir/atlasdb#743]
This code in the lock service reaper thread:

```
token = queue.take();
long sleepTimeMs = token.getExpirationDateMs() - currentTimeMillis() + maxAllowedClockDrift.toMillis();
if (sleepTimeMs > 0) {
    Thread.sleep(sleepTimeMs);
}
```

can result in locks not getting reaped in a timely manner. For example, if the token at the head of the queue has a 120 sec timeout, the thread will sleep for 120 sec. During that time, a lock request may come in with a much shorter timeout (e.g., 5 sec). However, that lock will not be reaped for 120 sec because the thread is asleep.

**Comment by gsheasby** [comment]
This looks like a one-line change, but the surrounding code is very messy, so we'd like to invest time in refactoring the class.

**Comment by gsheasby** [comment]
Related to #810

**Comment by pnepywoda** [comment]
@nziebart is this fixed now?

**Comment by nziebart** [comment]
Yes, fixed by #2041

**Comment by gsheasby** [comment]
Jolly good. Closing.
